pkg/types/gcp/validation/platform.go

```diff
@@ -53,6 +54,18 @@ func ValidatePlatform(p *gcp.Platform, fldPath *field.Path) field.ErrorList {
 	if _, ok := Regions[p.Region]; !ok {
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("region"), p.Region, validRegionValues))
 	}
+	if p.PublicZoneID != "" {
+		// PublicZoneID must be 1-63 characters long, must begin with a
+		// letter, end with a letter or digit, and only contain lowercase
+		// letters, digits or dashes.
+		match, err := regexp.MatchString("^[a-z][a-z0-9-]{0,61}[a-z0-9]?$", p.PublicZoneID)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("network"), p.PublicZoneID, err.Error()))
```

Comment:
the fldPath is incorrect here.

Reply:
personally, the regex patterns should be compilable, or else it's developer error and not user error.
we we should use https://golang.org/pkg/regexp/#MustCompile

and then use that to match using https://golang.org/pkg/regexp/#Regexp.MatchString

Reply:
Fixed.